python/hail/docs/hailpedia/matrixtable.rst

```diff
@@ -0,0 +1,404 @@
+fields, entry fields.
+
+In addition, there are different operations on the matrix for each field group.
+For instance, :class:`.Table` has :meth:`.Table.select` and
+:meth:`.Table.select_globals`, and :class:`.MatrixTable` has
+:meth:`.MatrixTable.select_rows`, :meth:`.MatrixTable.select_cols`,
+:meth:`.MatrixTable.select_entries`, and :meth:`.MatrixTable.select_globals`.
+
+It is possible to represent matrix data by coordinate in a table , storing one
+record per entry of the matrix. However, the :class:`.MatrixTable` represents
+this data far more efficiently and exposes natural interfaces for computing on
+it.
+
+The :meth:`.MatrixTable.rows` and :meth:`.MatrixTable.cols` methods return the
+row and column fields as separate tables. The :meth:`.MatrixTable.entries`
+method returns the matrix as a table in coordinate form -- use this object with
+caution.
+
+Keys
+====
+
+Matrix tables have keys just as tables do. However, instead of one key, matrix
+tables have two keys: a row key and a column key. Row fields are indexed by the
+row key, column fields are indexed by the column key, and entry fields are
+indexed by the row key and the column key. The key structs can be accessed with
+:attr:`.MatrixTable.row_key` and :attr:`.MatrixTable.col_key`. It is possible to
+change the key with :meth:`.MatrixTable.key_rows_by` and
+:meth:`.MatrixTable.key_cols_by`.
+
+Note that changing the row key, however, may be an expensive operation.
+
+Hail matrix tables are natively distributed objects, and as such have another
+key: a partition key. This key is used for specifying the ordering of the matrix
+table along the row dimension, which is important for performance. Access this
+with :attr:`.MatrixTable.partition_key`
+
+Referencing Fields
+==================
+
+All fields (row, column, global, entry) are top-level and exposed as attributes
+on the :class:`.MatrixTable` object. For example, if the matrix table `mt` had a
+row field `locus`, this field could be referenced with either ``mt.locus`` or
+``mt['locus']``. The former access pattern does not work with field names with
+spaces or punctuation.
+
+The result of referencing a field from a matrix table is an :class:`.Expression`
+which knows its type and knows its source as well as whether it is a row field,
+column field, entry field, or global field. Hail uses this context to know which
+operations are allowed for a given expression.
+
+When evaluated in a Python interpreter, we can see ``mt.locus`` is a
+:class:`.LocusExpression` with type ``locus<GRCh37>`` and it is a row field of
+the MatrixTable `mt`.
+
+    >>> mt
+    <hail.matrixtable.MatrixTable at 0x1107e54a8>
+
+    >>> mt.locus
+    <LocusExpression of type locus<GRCh37>>
+
+Likewise, ``mt.DP`` would be an :class:`.Int32Expression` with type ``int32``
+and is an entry field of `mt`. It is indexed by both rows and columns as denoted
+by its indices when describing the expression:
```

Comment:
"describe" is used off-hand here, but I think it's worth introducing it properly. Maybe:

> Hail expressions can also `describe` themselves, providing information on the source matrix table or table and which keys index the expression, if any. For example, ``mt.DP.describe()`` tells us that ``mt.DP`` has type ``int32`` and is an entry field of ``mt``:

(also mt should be double back-ticked)